app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt

```diff
@@ -168,15 +169,19 @@ enum class NavigationItem(
 
     OtherUserProfile(
         primaryRoute = OTHER_USER_PROFILE,
-        canonicalRoute = "$OTHER_USER_PROFILE/{$EXTRA_USER_DOMAIN}/{$EXTRA_USER_ID}",
+        canonicalRoute = "$OTHER_USER_PROFILE?$EXTRA_USER_ID={$EXTRA_USER_ID}",
+        deepLinks = listOf(navDeepLink {
+            uriPattern = "${DeepLinkProcessor.DEEP_LINK_SCHEME}://" +
+                    "${DeepLinkProcessor.OTHER_USER_PROFILE_DEEPLINK_HOST}/" +
+                    "{$EXTRA_USER_ID}"
+        }),
         content = { OtherUserProfileScreen() },
         animationConfig = NavigationAnimationConfig.NoAnimation
     ) {
         override fun getRouteWithArgs(arguments: List<Any>): String {
-            val userDomain: String = arguments.filterIsInstance<String>()[0]
-            val userProfileId: String = arguments.filterIsInstance<String>()[1]
-
-            return "$primaryRoute/$userDomain/$userProfileId"
+            val conversationIdString: String = arguments.filterIsInstance<QualifiedID>().firstOrNull()?.toString()
+                ?: "{$EXTRA_USER_ID}"
```

Comment:
Why if there is no such extra, then it passes "extra_user_id" string? I think it should be empty or just shouldn't set this argument at all

Reply:
in fact this is just "pretty" code, but anything that is not `QualifiedID` here will cause a crash in `OtherUserProfileViewModel`. 
Maybe good idea to throw an exception here, to have more informative logs 🤔